src/Domain/Services/DraftOrders.php

```diff
@@ -0,0 +1,151 @@
+			$this->maybe_remove_cronjobs();
+		}
+	}
+
+	/**
+	 * Installation related logic for Draft order functionality.
+	 */
+	public function install() {
+		$this->maybe_create_cronjobs();
+	}
+
+
+	/**
+	 * Remove cronjobs if they exist (but only from admin).
+	 */
+	protected function maybe_remove_cronjobs() {
+		if ( ! is_admin() ) {
+			return;
+		}
+		if ( function_exists( 'as_next_scheduled_action' ) && true === as_next_scheduled_action( 'woocommerce_cleanup_draft_orders' ) ) {
+			as_unschedule_all_actions( 'woocommerce_cleanup_draft_orders' );
+		}
+	}
+
+	/**
+	 * Maybe create cron events.
+	 */
+	protected function maybe_create_cronjobs() {
+		if ( function_exists( 'as_next_scheduled_action' ) && false === as_next_scheduled_action( 'woocommerce_cleanup_draft_orders' ) ) {
+			as_schedule_recurring_action( strtotime( 'midnight tonight' ), DAY_IN_SECONDS, 'woocommerce_cleanup_draft_orders' );
+		}
+	}
+
+
+	/**
+	 * Register custom order status for orders created via the API during checkout.
+	 *
+	 * Draft order status is used before payment is attempted, during checkout, when a cart is converted to an order.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @return array
+	 */
+	public static function register_draft_order_status( array $statuses ) {
+		$statuses['wc-checkout-draft'] = _x( 'Draft', 'Order status', 'woo-gutenberg-products-block' );
+		return $statuses;
+	}
+
+	/**
+	 * Register custom order post status for orders created via the API during checkout.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @return array
+	 */
+	public static function register_draft_order_post_status( array $statuses ) {
+		$statuses['wc-checkout-draft'] = [
+			'label'                     => _x( 'Draft', 'Order status', 'woo-gutenberg-products-block' ),
+			'public'                    => false,
+			'exclude_from_search'       => false,
+			'show_in_admin_all_list'    => false,
+			'show_in_admin_status_list' => true,
+			/* translators: %s: number of orders */
+			'label_count'               => _n_noop( 'Drafts <span class="count">(%s)</span>', 'Drafts <span class="count">(%s)</span>', 'woo-gutenberg-products-block' ),
+		];
+		return $statuses;
+	}
+
+	/**
+	 * Append draft status to a list of statuses.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @return array
+	 */
+	public static function append_draft_order_post_status( $statuses ) {
+		$statuses[] = 'checkout-draft';
+		return $statuses;
+	}
+
+	/**
+	 * Delete draft orders older than a day in batches of 20.
+	 *
+	 * Ran on a daily cron schedule.
+	 */
+	public static function delete_expired_draft_orders() {
+		$count      = 0;
+		$batch_size = 20;
+		$orders     = wc_get_orders(
+			[
+				'date_modified' => '<=' . strtotime( '-1 DAY' ),
+				'limit'         => $batch_size,
+				'status'        => 'wc-checkout-draft',
+				'type'          => 'shop_order',
+			]
+		);
+
+		if ( $orders ) {
+			foreach ( $orders as $order ) {
+				$order->delete( true );
```

Comment:
I was going to do this in a follow up but since we're moving all these methods anyway I think it should be here. We should wrap the delete in a `has_status` check to guard, and to be extra cautious I'd actually bail on the entire query if anything comes back unexpected. I'll think about this and suggest something.